packages/relayer/src/provider-relayer.ts

```diff
@@ -91,7 +104,61 @@ export abstract class ProviderRelayer extends BaseRelayer implements Relayer {
     return (await module.readNonce(space ? space : 0, { blockTag: blockTag })).toNumber()
   }
 
-  wait(_metaTxnId: any, _timeout: number): Promise<any> {
-    throw new Error('Method not implemented.')
+  async wait(metaTxnId: string | SignedTransactions, timeout: number): Promise<providers.TransactionResponse & providers.TransactionReceipt> {
+    if (typeof metaTxnId !== 'string') {
+      console.log("computing id", metaTxnId.config, metaTxnId.context, metaTxnId.chainId, ...metaTxnId.transactions)
+      return this.wait(
+        computeMetaTxnHash(addressOf(metaTxnId.config, metaTxnId.context), metaTxnId.chainId, ...metaTxnId.transactions),
+        timeout
+      )
+    }
+  
+    // Transactions can only get executed on nonce change
+    // get all nonce changes and look for metaTxnIds in between logs
+    const timeoutTime = new Date().getTime() + timeout
+    let lastBlock: number = 0
+
+    const normalMetaTxnId = metaTxnId.replace('0x', '')
+
+    while (new Date().getTime() < timeoutTime) {
+      const block = await this.provider.getBlockNumber()
+      const logs = await this.provider.getLogs({
+        fromBlock: Math.max(0, lastBlock - 12),
+        toBlock: block,
+        // Nonce change event topic
+        topics: ['0x1f180c27086c7a39ea2a7b25239d1ab92348f07ca7bb59d1438fcf527568f881']
+      })
+
+      lastBlock = block
+
+      // Get receipts of all transactions
+      const txs = await Promise.all(logs.map((l) => this.provider.getTransactionReceipt(l.transactionHash)))
+
+      // Find a transaction with a TxExecuted log
+      const found = txs.find((tx) => tx.logs.find((l) => (
+        (
+          l.topics.length === 0 &&
+          l.data.replace('0x', '') === normalMetaTxnId
+        ) || (
+          l.topics.length === 1 &&
+          // TxFailed event topic
+          l.topics[0] === "0x3dbd1590ea96dd3253a91f24e64e3a502e1225d602a5731357bc12643070ccd7" &&
+          l.data.length >= 64 && l.data.replace('0x', '').startsWith(normalMetaTxnId) 
+        )
+      )))
+
+      // If found return that
+      if (found) {
+        return {
+          ...found,
+          ...await this.provider.getTransaction(found.transactionHash)
+        }
+      }
+
+      // Otherwise wait and try again
+      await new Promise(r => setTimeout(r, 1000))
```

Comment:
Should be using the waitPollRate variable here, not hardcoding 1000

Reply:
Yep, it was an oversight, I just fixed it :sweat_smile: